ts/javascript/mod.ts

```diff
@@ -39,7 +39,7 @@ const fromNodeVersion = (
   const { pkg, permittedInsecurePackages } = nodeVersions[version];
   return {
     pkgs: nixRaw`
-      import "\${nixpkgs}" {
+      import "\${nixpkgs-repo}" {
```

Comment:
Currently we declare `nixpkgs` [here](https://github.com/garnix-io/garn/blob/966fe4114d091aff39c302d9a21f29a44d6d9303/ts/internal/runner.ts#L143) and then use that in other places (for example here). That's not ideal, since it's an implicit dependency. But why is changing it to `nixpkgs-repo` an improvement? Then it implicitly depends on someone somewhere else using `nixFlakeDep("nixpkgs-repo", ...`. And that seems even more implicit.

I think at some point we want to move to using something like `nixFlakeDep` whereever we need `nixpkgs`. But it's a little bit more involved to do that. So I wouldn't do that in this PR.

Or do you need this change for anything?

Reply:
FormatFlake introduces 'nixpkgs' via a let binding, but buildPackage doesn't. Therefore I did this change to make mkNpmProject compatible to buildPackage. As disvussed in discourse yesterday, we should probably fix this via a smarter nixFlakeDep that deduplicates. I could do this via another PR.

Reply:
I guess I can just make use of nixFlakeDep already here, then take care of deduplication later.

Reply:
Oh, right. I thought `buildPackage` would work without changing this, sorry.

I would maybe try to make `buildPackage` mimic what `formatFlake` does then. So that the tests are closer to what's actually going on. For example like this: f641ee33

Reply:
OK, I now made mkNpmProject use nixFlakeDep to refer to the flake input to be a bit more explicit about it.
I also added the `nixpkgs` and `system` bindings to the test helpers.